### PR TITLE
Convert various to callnatives - Part 1

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
First batch of conversions.

- handleultraburst
- handleprimalreversion
- handlemegaevo
- jumpifshelltrap
- jumpifemergencyexited
- jumpifrodaffected
- jumpifabsorbaffected
- jumpifmotoraffected

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: src/battle_script_commands.c
Comment (ghoulslash) [inline src/battle_script_commands.c:16325]: in preparation for #3262 let's use a local variable instead of gActiveBattler

File: src/battle_script_commands.c
Comment (ghoulslash) [inline src/battle_script_commands.c:16327]: might be useful to have some macros e.g. `#define NEXT_INSTR  gBattlescriptCurrInstr = cmd->nextInstr` and a similar FAIL_INSTR so we can put that at the end of these BS callnatives

**Review by ghoulslash — APPROVED** [review]

**Review by ghoulslash — CHANGES_REQUESTED** [review]
see previous comments

File: src/battle_script_commands.c
Comment (AsparagusEduardo) [reply src/battle_script_commands.c:16325]: Weird thing is, I initially tried using a local variable but for somr reason it doubled the messages when when doing it for Ultra Burst, which is the first one I tried. I'll try to find the source of the issue once I get time.

Comment (ghoulslash) [reply src/battle_script_commands.c:16325]: Maybe there was some downstream dependence on gActiveBattler. Let's hold off until that pr gets merged

Comment (AsparagusEduardo) [reply src/battle_script_commands.c:16325]: Sure, though the current test didn't pick up on the double message, so that's something to check with that PR as well.

Comment (AsparagusEduardo) [reply src/battle_script_commands.c:16325]: This was indeed the case, double messags are gone after the merge with upcoming.

File: src/battle_script_commands.c
Comment (AsparagusEduardo) [reply src/battle_script_commands.c:16327]: Since `gBattlescriptCurrInstr = cmd->nextInstr` is used by all 3 `various`, `callnative` and regular commands, it might be better to have this as a separate PR.

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**
